cli/src/action/registry/mod.rs

```diff
@@ -177,3 +182,161 @@ impl Action for RegistryGenerateAction {
         Ok(())
     }
 }
+
+#[cfg(feature = "registry-add-node")]
+pub struct RegistryAddAction;
+
+#[cfg(feature = "registry-add-node")]
+impl Action for RegistryAddAction {
+    fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
+        let args = arg_matches.ok_or(CliError::RequiresArgs)?;
+
+        let url = args
+            .value_of("url")
+            .map(ToOwned::to_owned)
+            .or_else(|| std::env::var(SPLINTER_REST_API_URL_ENV).ok())
+            .unwrap_or_else(|| DEFAULT_SPLINTER_REST_API_URL.to_string());
+
+        let identity = args
+            .value_of("identity")
+            .ok_or_else(|| CliError::ActionError("Identity must be specified".into()))?
+            .to_string();
+
+        let display_name = args
+            .value_of("display_name")
+            .unwrap_or(&identity)
+            .to_string();
```

Comment:
I don't expect this change in this PR, but at some point we should probably update the registry REST API to accept a new node without the display name field set; the REST API would then generate a default value for this field internally. Can you create a doc for this feature add this as a change before we stabilize?

Reply:
Created a doc with this comment, will add it to the feature stabilization list